src/terminal/adapter/adaptDispatchGraphics.cpp

```diff
@@ -260,20 +320,30 @@ size_t AdaptDispatch::_ApplyGraphicsOption(const VTParameters options,
 }
 
 // Routine Description:
-// - This is a no-op until we have something meaningful to do with sub parameters.
 // - Helper to apply a single graphic rendition option with sub parameters to an attribute.
 // Arguments:
 // - option - An option to apply.
+// - subParams - Sub parameters associated with the option.
 // - attr - The attribute that will be updated with the applied option.
 // Return Value:
 // - <None>
-void AdaptDispatch::_ApplyGraphicsOptionSubParam(const VTParameter /* option */,
-                                                 const VTSubParameters /* subParam */,
-                                                 TextAttribute& /* attr */) noexcept
+void AdaptDispatch::_ApplyGraphicsOptionAlt(const VTParameter option,
+                                            const VTSubParameters subParams,
+                                            TextAttribute& attr) noexcept
 {
     // here, we apply our "best effort" rule, while handling sub params if we don't
     // recognise the parameter substring (parameter and it's sub parameters) then
     // we should just skip over them.
+    switch (option)
+    {
+    case ForegroundExtended:
+    case BackgroundExtended:
+        _SetRgbColorsHelperAlt(option, subParams, attr);

```

Comment:
Instead of passing the `option` through to `_SetRgbColorsHelperAlt`, would it perhaps be easier to pass through the `isForeground` state as a parameter instead, the same way the `_SetRgbColorsHelper` method works? So this switch would look something like this:

```cpp
case ForegroundExtended:
    _SetRgbColorsHelperAlt(subParams, attr, true);
    break;
case BackgroundExtended:
    _SetRgbColorsHelperAlt(subParams, attr, false);
    break
```

Reply:
That's a little future thing. We can use the same helper to extract color for underlines when we support `SGR 58`.

Reply:
OK. That makes sense.